cmake/HPX_SetupBoost.cmake

```diff
@@ -5,6 +5,65 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
+if(HPX_WITH_FETCH_BOOST)
+  set(HPX_WITH_BOOST_VERSION "1.84.0")
+  hpx_info(
+    "HPX_WITH_FETCH_BOOST=${HPX_WITH_FETCH_BOOST}, Boost v${HPX_WITH_BOOST_VERSION} will be fetched using CMake's FetchContent"
+  )
+  include(FetchContent)
+  fetchcontent_declare(
+    Boost
+    URL https://github.com/boostorg/boost/releases/download/boost-${HPX_WITH_BOOST_VERSION}/boost-${HPX_WITH_BOOST_VERSION}.tar.gz
+    TLS_VERIFY true
+    DOWNLOAD_EXTRACT_TIMESTAMP true
+  )
+  fetchcontent_populate(Boost)
+
+  set(EX_PRC
+      "Execute process"
+      CACHE STRING "Used by command line tool."
+  )
+  set(EX_PRC_INTERNAL
+      ""
+      CACHE INTERNAL "for internal use only; do not modify"
+  )
+
+  if(NOT EX_PRC STREQUAL EX_PRC_INTERNAL)
+    if(WIN32)
+      execute_process(
+        COMMAND
+          cmd /C
+          "cd ${CMAKE_BINARY_DIR}\\_deps\\boost-src && .\\bootstrap.bat && .\\b2 headers cxxflags=/std:c++20"
```

Comment:
Should we compile with the C++ version that corresponds to what the user requested for HPX (`HPX_CXX_STANDARD`)?

Reply:
Yes, we could do that. Will make the changes